File: Lib/Java/com.scbeta.net/src/com/scbeta/net/xiep/packages/ResponsePackage.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.scbeta.net.xiep.packages;

import com.scbeta.helpers.XmlTreeNode;

/**
 * 响应包
 * @author aaa
 */
public class ResponsePackage extends AbstractXiepPackage {

    @Override
    public String getPackageName() {
        return "ResponsePackage";
    }

    //获取请求编号
    public String getRequestId() {
        return this.getAttributeValue("RequestId");
    }

    //设置请求编号
    public void setRequestId(String value) {
        this.setAttribute("RequestId", value);
    }

    //获取响应
    public String getResponse() {
        return this.getAttributeValue("Response");
    }

    //设置响应
    public void setResponse(String value) {
        this.setAttribute("Response", value);
    }

    //构造函数
    public ResponsePackage() {
    }

    //构造函数
    public ResponsePackage(String requestId) {
        init(requestId, null);
    }

    //构造函数
    public ResponsePackage(String requestId, String response) {
        init(requestId, response);
    }

    //构造函数
    public ResponsePackage(RequestPackage requestPackage) {
        init(requestPackage.getRequestId(), null);
    }

    //构造函数
    public ResponsePackage(RequestPackage requestPackage, String response) {
        init(requestPackage.getRequestId(), response);
    }

    private void init(String requestId, String response) {
        setRequestId(requestId);
        if (response == null || response.isEmpty()) {
        } else {
            this.setResponse(response);
        }
    }

    //从XML得到响应包
    public static ResponsePackage fromXml(String xml) {
        ResponsePackage rtnPackage = null;

        XmlTreeNode treeNode = XmlTreeNode.fromXml(xml);
        String packageName = treeNode.getKey();
        if (packageName.equals("ResponsePackage")) {
            rtnPackage = new ResponsePackage();
        } else {
            return null;
        }

        rtnPackage.setRootPackage(treeNode);
        return rtnPackage;
    }
}

File: trunk/Lib/DotNet/aaaSoft.Net/aaaSoft.Net/Xiep/Packages/ResponsePackage.cs
using System;
using System.Collections.Generic;
using System.Text;
using aaaSoft.Helpers;

namespace aaaSoft.Net.Xiep.Packages
{
    public class ResponsePackage:AbstractXiepPackage
    {
        public override string GetPackageName()
        {
            return "ResponsePackage";
        }

        //获取或设置请求编号
        public String RequestId
        {
            get { return this.GetAttributeValue("RequestId"); }
            set { this.SetAttribute("RequestId", value); }
        }

        //获取或设置响应
        public String Response
        {
            get { return this.GetAttributeValue("Response"); }
            set { this.SetAttribute("Response", value); }
        }

        //构造函数
        public ResponsePackage()
        {
        }

        //构造函数
        public ResponsePackage(String requestId)
        {
            init(requestId, null);
        }

        //构造函数
        public ResponsePackage(String requestId, String response)
        {
            init(requestId, response);
        }

        //构造函数
        public ResponsePackage(RequestPackage requestPackage)
        {
            init(requestPackage.RequestId, null);
        }

        //构造函数
        public ResponsePackage(RequestPackage requestPackage, String response)
        {
            init(requestPackage.RequestId, response);
        }

        private void init(String requestId, String response)
        {
            RequestId = requestId;
            if (String.IsNullOrEmpty(response))
            {
            }
            else
            {
                this.Response = response;
            }
        }

        //从XML得到响应包
        public static ResponsePackage fromXml(String xml)
        {
            ResponsePackage rtnPackage = null;

            XmlTreeNode treeNode = XmlTreeNode.FromXml(xml);
            String packageName = treeNode.Key;
            if (packageName.Equals("ResponsePackage"))
            {
                rtnPackage = new ResponsePackage();
            }
            else
            {
                return null;
            }

            rtnPackage.SetRootPackage(treeNode);
            return rtnPackage;
        }
    }
}
